### DISCUSSION TITLE
Download blog in as-is state

### DISCUSSION THREAD

**Discussion by Sylvan-Knight** [TumblThreeApp/TumblThree#287] [Ideas]
**Is your feature request related to a problem? Please describe.**
I would like to download a blog so it is similar to an offline mirror of the blog.

**Describe the solution you'd like**
An html page that can be opened and scrolled through, linked to the pictures and text, so that pictures and videos have their text positioned with them for context.

**Describe alternatives you've considered**
Something akin to screenshots or gifs that are expanded beyond the scope of the picture so that it captures text and tags accompanying it.

**Additional context**
A full viewable actual backup rather than just saving the pictures.

**Comment by hrxn** [comment]
https://github.com/Cebtenzzre/tumblr-utils

**Comment by thomas694** [comment]
Thanks for your feature suggestion.
We'll think about it. The change required for this would require quite a few more changes and will probably not come too soon because the available development resources are low at the moment. Volunteers are welcome.

**Comment by jacob-pro** [comment]
I think a middle solution would be to ensure that TumblThree saves accurate and useful metadata in the images.txt etc. then it is possible to write a simple frontend viewer that serves the archived files ?
https://github.com/TumblThreeApp/TumblThree/issues/229

**Comment by jacob-pro** [comment]
I have written a [tool](https://github.com/jacob-pro/tumbl-three-viewer) that renders your downloaded TumblThree blogs in your browser.

It just requires that you have enabled the JSON metadata files 

![image](https://user-images.githubusercontent.com/16433349/231573282-a447ba84-5722-4446-a545-16218735277f.png)

Download here:  https://github.com/jacob-pro/tumbl-three-viewer/releases
